pkg/labels/http_applicator.go

```diff
@@ -0,0 +1,80 @@
+package labels
+
+import (
+	"encoding/json"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+	"io/ioutil"
+	"net/http"
+)
+
+type httpApplicator struct {
+	client *http.Client
+	// The endpoint that will be queried for matches.
+	// GetMatches appends selector.String() to the end of this string.
+	matchesEndpoint string
+	logger          logging.Logger
+}
+
+var _ Applicator = &httpApplicator{}
+
+func NewHttpApplicator(client *http.Client, matchesEndpoint string) *httpApplicator {
+	c := client
+	if c == nil {
+		c = http.DefaultClient
+	}
+
+	return &httpApplicator{
+		logger:          logging.DefaultLogger,
+		matchesEndpoint: matchesEndpoint,
+		client:          c,
+	}
+}
+
+func (h *httpApplicator) SetLabel(labelType Type, id, name, value string) error {
+	return util.Errorf("SetLabel not implemented for HttpApplicator (type %s, id %s, name %s, value %s)", labelType, id, name, value)
+}
+
+func (h *httpApplicator) RemoveLabel(labelType Type, id, name string) error {
+	return util.Errorf("RemoveLabel not implemented for HttpApplicator (type %s, id %s, name %s)", labelType, id, name)
+}
+
+func (h *httpApplicator) GetLabels(labelType Type, id string) (Labeled, error) {
+	return Labeled{}, util.Errorf("GetLabels not implemented for HttpApplicator (type %s, id %s)", labelType, id)
+}
+
+func (h *httpApplicator) GetMatches(selector Selector, labelType Type) ([]Labeled, error) {
+	resp, err := h.client.Get(h.matchesEndpoint + selector.String())
+	if err != nil {
+		return []Labeled{}, err
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return []Labeled{}, err
+	}
+
+	matches := []interface{}{}
+
+	err = json.Unmarshal(body, &matches)
+	if err != nil {
+		return []Labeled{}, err
+	}
+
+	labeled := make([]Labeled, len(matches))
+
+	for i, v := range matches {
+		s, ok := v.(string)
+		if !ok {
+			return []Labeled{}, util.Errorf("when parsing %s: %+v not a string", body, v)
+		}
+		labeled[i] = Labeled{
+			ID:        s,
+			LabelType: labelType,
+			Labels:    Set{},
```

Comment:
I can adjust the backend implementation to return the labels on these objects if you prefer